### PR TITLE
feature-khweb-8/add-events-page

### PR DESCRIPTION
**Pull request by topsart**
[Add events page](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=3370&view=detail&selectedIssue=KHWEB-8)

- added page structure
- added styles
- added redux

### CODE REVIEW

File: src/components/EventsPage/api/services/getEventsData.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Dispatch } from 'react';
```
Comment (PietyE) [inline src/components/EventsPage/api/services/getEventsData.ts:1]: add Dispatch from redux and add type

File: src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemContent/EventsItemContent.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  date,
+}: EventsItemContentProps) => {
+  const eventDate = moment(date).utc().format('LLLL');
+  let slicedDesription = '';
+
+  if (description.length > 145) {
+    slicedDesription = description.substr(0, 145) + '...';
+  } else {
+    slicedDesription = description;
+  }
```
Comment (PietyE) [inline src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemContent/EventsItemContent.tsx:24]: Typography have props noWrap={true} , you can use it there

File: src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemActions/EventsItemActions.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const EventsItemActions: React.FC<EventsItemActionsProps> = ({
+  video,
+}: EventsItemActionsProps) => (
+  <CardActions className="events-item-actions">
+    {!video ? (
+      <Typography className="events-item-actions-info" variant="body1">
+        No Video Available
+      </Typography>
```
Comment (kerrimov) [inline src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemActions/EventsItemActions.tsx:17]: hide button if video unavailable

File: src/components/EventsPage/api/services/getEventsData.ts (hunk 1 above)
Comment (topsart) [reply src/components/EventsPage/api/services/getEventsData.ts:1]: fixed

File: src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemActions/EventsItemActions.tsx (hunk 3 above)
Comment (topsart) [reply src/components/EventsPage/components/EventsList/components/EventsItem/components/EventsItemActions/EventsItemActions.tsx:17]: done

**Review by kerrimov — APPROVED** [review]

**Pull request merged**